polkadot-parachains/parachains-common/src/xcm_config.rs

```diff
@@ -0,0 +1,67 @@
+use core::marker::PhantomData;
+use frame_support::{log, weights::Weight};
+use xcm::latest::prelude::*;
+use xcm_executor::traits::ShouldExecute;
+
+//TODO: move DenyThenTry to polkadot's xcm module.
+/// Deny executing the XCM if it matches any of the Deny filter regardless of anything else.
+/// If it passes the Deny, and matches one of the Allow cases then it is let through.
+pub struct DenyThenTry<Deny, Allow>(PhantomData<Deny>, PhantomData<Allow>)
+where
+	Deny: ShouldExecute,
+	Allow: ShouldExecute;
+
+impl<Deny, Allow> ShouldExecute for DenyThenTry<Deny, Allow>
+where
+	Deny: ShouldExecute,
+	Allow: ShouldExecute,
+{
+	fn should_execute<Call>(
+		origin: &MultiLocation,
+		message: &mut Xcm<Call>,
+		max_weight: Weight,
+		weight_credit: &mut Weight,
+	) -> Result<(), ()> {
+		Deny::should_execute(origin, message, max_weight, weight_credit)?;
+		Allow::should_execute(origin, message, max_weight, weight_credit)
+	}
+}
+
+// See issue #5233
+pub struct DenyReserveTransferToRelayChain;
```

Comment:
May I ask your reasoning for introducing this filter here? Because for HRMP we actually are using the relay chain for reserve transfer.

Reply:
Please check https://github.com/paritytech/polkadot/issues/5233. We are not forced to use this filter.
However, I have changed the barrier's config, which could be wrong. https://github.com/integritee-network/parachain/blob/6007056933c2893d490def06e8533af6f2e83e5a/polkadot-parachains/integritee-runtime/src/xcm_config.rs#L226
What is our goal?

Reply:
Our Barrier is certainly not yet perfect, but I'm not sure how useful this barrier actually will be. Because a `DepositReserveAsset` on our chain to the relay chain account does not make any sense, that is true. But neither would it make sense to any parachain account... nor do we actually expect ReserveTransfers on our chain I believe?

I'm suggesting to remove this filter for now. If we want to try protecting users from loosing TEER by depositing them onto inaccessible accounts, we will have to put up a more thorough barrier, which should be an issue itself.

Reply:
I guess we should keep this filter because:

1. We don't need to transfer assets to the relay chain anyhow
2. Reading this https://github.com/paritytech/polkadot/issues/5233, I suppose that it would look funds that users can't recover.